dbms/src/DataStreams/WindowTransformAction.cpp

```diff
@@ -660,7 +660,7 @@ std::tuple<RowNumber, bool> WindowTransformAction::stepToStartForRangeFrame()
 
 std::tuple<RowNumber, bool> WindowTransformAction::stepToEndForRangeFrame()
 {
-    if (!window_description.frame.end_preceding && !partition_ended)
+    if (!partition_ended)
         // If we find the frame end and the partition_ended is false.
```

Comment:
NOTE: for `expr` in `expr preceding`, `expr` can only be positive number, so no need to change `stepToStartForRangeFrame`